metron-interface/metron-rest/src/main/java/org/apache/metron/rest/service/impl/StormStatusServiceImpl.java

```diff
@@ -145,4 +136,35 @@ protected String getStormUiProperty() {
     }
     return baseValue;
   }
+
+  /**
+   * Retrieves the Storm topology id from the given topology name.  If a topology name is detected to be an aggregate
+   * parser topology, the SensorParserGroups are checked for a match.
+   * @param name Topology or SensorParserGroup name
+   * @return Topology id
+   */
+  protected String getTopologyId(String name) {
+    String id = null;
+    for (TopologyStatus topology : getTopologySummary().getTopologies()) {
+      String topologyName = topology.getName();
+
+      // check sensor group
+      if (topologyName.contains(ParserTopologyCLI.STORM_JOB_SEPARATOR)) {
+        Set<String> sensors = new HashSet<>(Arrays.asList(topologyName.split(ParserTopologyCLI.STORM_JOB_SEPARATOR)));
```

Comment:
Is this related to the code that is in `StormAdminServiceImpl.startParserTopology` where it creates the topology name from the sensor types?   This is just the inverse, right?

It seems like all logic that maps sensors to a topology name and vice-versa, could exist in a single class that is reused.  This would let us easily throw a bunch of unit tests around it  and ensure we catch any weird corner cases.

Reply:
The code in `StormAdminServiceImpl.startParserTopology` is creating the `ParserTopologyCLI` sensor type option so it's not quite the inverse.  The `StormStatusServiceImpl.getTopologyId` method is only used in the same class and the `StormAdminServiceImpl.startParserTopology` is duplicated, but the duplication is in Ambari python code.

How do you think we should organize it?

Reply:
If there is not an obvious way to do it, then let's not.  Thanks for taking a look.